xc/common/primitives/ff/ff.pb_type.xml

```diff
@@ -523,7 +523,10 @@
       <direct   name="Q"        input="LDPE_or_LDCE.Q"   output="SLICE_FF.Q"      />
     </interconnect>
     <metadata>
-      <meta name="fasm_features">LATCH</meta>
+      <meta name="fasm_features">
+        LATCH
+        CLKINV
```

Comment:
If you need that feature enabled to make latches work then there is a need of a separate mode for the whole COMMON_SLICE. This feature is slice-wide.

I imagine you could make COMMON_SLICE have two major modes eg. "POSEDGE" and "NEGEDGE". The former will contain sites for posedge FFs while the latter for negedge FFs and latches.

Reply:
Actually, both LATCH and CLKINV are features that are already set at the SLICE level, this means that all the FFs will be in LATCH mode (with the clock not inverted thanks to the CLKINV feature).

Reply:
Ok, so then we will need a separate COMMON_SLICE mode for latch only.

Reply:
I don't think this is needed as well. the clock is used only for the FF portion of the slice. The mode selecting the latch types is already defined here: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/0e9281577f10ba2e2688c1de51e45eaa81302dea/xc/common/primitives/ff/ff.pb_type.xml#L444-L528.

Basically, when there is a latch primitive, all of the FF of a slice are bound to be in that mode. So, all the modes governing the packing ways of the FF resources are all defined in the [ff.pb_type.xml](https://github.com/SymbiFlow/symbiflow-arch-defs/blob/0e9281577f10ba2e2688c1de51e45eaa81302dea/xc/common/primitives/ff/ff.pb_type.xml)

Reply:
The clock is used in the LATCH mode?  It is the "G" input?  But yet, the LATCH feature is site wide.

Reply:
Yeah, clock is the G input for latches

Reply:
Shouldn't `CLKINV` be controlled by `not(IS_G_INVERTED)`?  At a minimum, we should add a TODO here?